Makefile

```diff
@@ -16,21 +16,25 @@ BIN_DIR           ?= $(FIRST_GOPATH)/bin
 DEP_FINISHED      ?= .dep-finished
 
 # Tools.
-DEP               ?= $(BIN_DIR)/dep-$(DEP_VERSION)
-DEP_VERSION       ?= 45be32ba4708aad5e2aa8c86f9432c4c4c1f8da2
-EMBEDMD           ?= $(BIN_DIR)/embedmd-$(EMBEDMD_VERSION)
+DEP                 ?= $(BIN_DIR)/dep-$(DEP_VERSION)
+DEP_VERSION         ?= 45be32ba4708aad5e2aa8c86f9432c4c4c1f8da2
+EMBEDMD             ?= $(BIN_DIR)/embedmd-$(EMBEDMD_VERSION)
 # v2.0.0
-EMBEDMD_VERSION   ?= 97c13d6e41602fc6e397eb51c45f38069371a969
-ERRCHECK          ?= $(BIN_DIR)/errcheck-$(ERRCHECK_VERSION)
+EMBEDMD_VERSION     ?= 97c13d6e41602fc6e397eb51c45f38069371a969
+ERRCHECK            ?= $(BIN_DIR)/errcheck-$(ERRCHECK_VERSION)
 # v1.2.0
-ERRCHECK_VERSION  ?= e14f8d59a22d460d56c5ee92507cd94c78fbf274
-LICHE             ?= $(BIN_DIR)/liche-$(LICHE_VERSION)
-LICHE_VERSION     ?= 2a2e6e56f6c615c17b2e116669c4cdb31b5453f3
-GOIMPORTS         ?= $(BIN_DIR)/goimports-$(GOIMPORTS_VERSION)
-GOIMPORTS_VERSION ?= 1c3d964395ce8f04f3b03b30aaed0b096c08c3c6
-PROMU             ?= $(BIN_DIR)/promu-$(PROMU_VERSION)
+ERRCHECK_VERSION    ?= e14f8d59a22d460d56c5ee92507cd94c78fbf274
+LICHE               ?= $(BIN_DIR)/liche-$(LICHE_VERSION)
+LICHE_VERSION       ?= 2a2e6e56f6c615c17b2e116669c4cdb31b5453f3
+GOIMPORTS           ?= $(BIN_DIR)/goimports-$(GOIMPORTS_VERSION)
+GOIMPORTS_VERSION   ?= 1c3d964395ce8f04f3b03b30aaed0b096c08c3c6
+PROMU               ?= $(BIN_DIR)/promu-$(PROMU_VERSION)
 # v0.2.0
-PROMU_VERSION     ?= 264dc36af9ea3103255063497636bd5713e3e9c1
+PROMU_VERSION       ?= 264dc36af9ea3103255063497636bd5713e3e9c1
+PROTOC              ?= $(BIN_DIR)/protoc-$(PROTOC_VERSION)
+PROTOC_VERSION      ?= 3.4.0
+PROTOC_PACKAGE      ?= protoc-$(PROTOC_VERSION)-linux-x86_64.zip
```

Comment:
I would put it to golang `BIN_DIR` if possible (: What do you think?

Reply:
em, actually it's possible in theory, but the package has sub `bin` dir, I should extract the binary only. That is why I put the package at the top of temporary dir.

Reply:
what do you mean? You can use PATH to file.. it's even better

Reply:
It's actually important, otherwise users can accidently commit this file

Reply:
oh.. Yes I totally missed that! no... TMP is totally fine for protoc package..! sorry